### PR TITLE
Mosek: Accept Unknown

### PR DESCRIPTION
**Pull request by Paulnkk**
## Description
I added a parameter for the mosek solver option to map solver status 'unknown' to an inaccurate solution if applicable.
Issue link (if applicable):
I found the issue here: https://github.com/cvxpy/cvxpy/issues/1787

## Type of change
- [ ] New feature (backwards compatible)
- [ ] New feature (breaking API changes)
- [x] Bug fix
- [ ] Other (Documentation, CI, ...)

## [Contribution checklist](https://www.cvxpy.org/contributing/index.html#contribution-checklist)
- [ ] Add our license to new files.
- [x] Check that your code adheres to our coding style.
- [ ] Write unittests.
- [ ] Run the unittests and check that they’re passing.
- [ ] Run the benchmarks to make sure your change doesn’t introduce a regression.

### CODE REVIEW

File: cvxpy/reductions/solvers/conic_solvers/mosek_conif.py [hunk 1]
```diff
@@ -469,6 +469,18 @@ def invert(self, solver_output, inverse_data):
                       mosek.solsta.prim_feas: s.OPTIMAL_INACCURATE,    # for integer problems
                       mosek.solsta.prim_infeas_cer: s.INFEASIBLE,
                       mosek.solsta.dual_infeas_cer: s.UNBOUNDED}
+
+        solver_opts = solver_output['solver_options']
+        # man muss hier über
```
Comment (phschiele) [inline cvxpy/reductions/solvers/conic_solvers/mosek_conif.py:474]: :de: comment :)

File: cvxpy/reductions/solvers/conic_solvers/mosek_conif.py [hunk 2]
```diff
@@ -469,6 +469,18 @@ def invert(self, solver_output, inverse_data):
...
+        if 'accept_unknown' in solver_opts:
+            if solver_opts['accept_unknown']:
```
Comment (phschiele) [inline cvxpy/reductions/solvers/conic_solvers/mosek_conif.py:476]: Can be combined as `if solver_opts.get('accept_unknown', False)`

File: cvxpy/reductions/solvers/conic_solvers/mosek_conif.py [hunk 3]
```diff
@@ -469,6 +469,18 @@ def invert(self, solver_output, inverse_data):
...
+                STATUS_MAP[mosek.solsta.unknown] = s.OPTIMAL_INACCURATE
+                STATUS_MAP[mosek.solsta.optimal] = s.OPTIMAL
+                STATUS_MAP[mosek.solsta.integer_optimal] = s.OPTIMAL
+                STATUS_MAP[mosek.solsta.prim_feas] = s.OPTIMAL_INACCURATE
+                STATUS_MAP[mosek.solsta.prim_infeas_cer] = s.INFEASIBLE
+                STATUS_MAP[mosek.solsta.dual_infeas_cer] = s.UNBOUNDED
```
Comment (phschiele) [inline cvxpy/reductions/solvers/conic_solvers/mosek_conif.py:482]: We would only need to update the entries that changed from above.

**Comment by phschiele** [comment]
@Paulnkk Thanks for adding this! 
In addition to a few small comments above, we should also document the setting here:
https://www.cvxpy.org/tutorial/advanced/index.html#setting-solver-options

It would be even better if we could add a test case. Paging @pqrz, @aszekMosek, @erling-d-andersen, @rileyjmurray , and @SteveDiamond from the discussion in #1787 for ideas about a minimal example that can trigger the unknown status (with solution).

**Comment by rileyjmurray** [comment]
@phschiele: a few months ago (between 1.2 and 1.3 releases) we found an example in ``test_constant_atoms`` that resulted in an ill-posed exponential cone program. I think the exponential cone is the only one that MOSEK supports which is defined as a closure of some inequality-representable set. So I would try making such an example where the only feasible solutions lay on the boundary that needs the closure.

File: cvxpy/reductions/solvers/conic_solvers/mosek_conif.py (hunk 2 above)
Comment (Paulnkk) [reply cvxpy/reductions/solvers/conic_solvers/mosek_conif.py]: Merged conditions together here: [Adjust accept_unknown condition](https://github.com/cvxpy/cvxpy/pull/2117/commits/abf94f70f559daef24aacb5f1efa1257e319d318)

File: cvxpy/reductions/solvers/conic_solvers/mosek_conif.py (hunk 3 above)
Comment (Paulnkk) [reply cvxpy/reductions/solvers/conic_solvers/mosek_conif.py]: Removed conditions here: [Adjust accept_unknown condition](https://github.com/cvxpy/cvxpy/pull/2117/commits/abf94f70f559daef24aacb5f1efa1257e319d318)

**Comment by aszekMosek** [comment]
To reliably get solution status unknown take any problem that is nontrivial enough not to get solved by presolve and set time limit or max number of iterations to 0.

**Comment by Paulnkk** [comment]
@phschiele @rileyjmurray @aszekMosek I added a problem from a simple cone program which already exists in cvxpy with setting timelimit to 0 but I need to double check if the problem is actually non-trivial enough to to trigger the unknown status and that the assertion I build actually leads to the desired check we want here.

**Comment by rileyjmurray** [comment]
I think it suffices to check that the status is "optimal - inaccurate".
With this PR it's clear that we send that status back only when MOSEK
returns an "unknown" status code. If MOSEK's presolve gets better in the
future we may need to update the test problem. I think we can cross that
bridge if / when we come to it.

**Comment by Paulnkk** [comment]
@rileyjmurray thanks, I think I implemented it with the desired output but termination with OPTIMAL_INACCURATE is actually trivial since we have no timelimit.

**Review by rileyjmurray — CHANGES_REQUESTED** [review]
@Paulnkk can you please update the web documentation to mention this parameter?

The relevant section that needs to change is here: https://github.com/cvxpy/cvxpy/blob/master/doc/source/tutorial/advanced/index.rst#setting-solver-options.

**Review by rileyjmurray — APPROVED** [review]

**Review by SteveDiamond — APPROVED** [review]

**Review by phschiele — APPROVED** [review]

**Pull request merged**